RELEASE.md

```diff
@@ -1,23 +1,104 @@
+`1.0.0-rc.2` is a pre-release. It is the second release candidate before
+releasing the final `1.0.0`.
+
+A release branch has the name structure `release-X.Y` where `X.Y` stands for the
+minor version. Releases like `X.Y.Z` will all be released from the branch
+`release-X.Y`.
+
+For example, `1.0.2` release should be on `release-1.0` branch.
+
+## Contributor Instructions
+
+The following section targets contributors. No need to read further if you only
+want to use Kubeflow Pipelines.
+
+### Cherry picking pull requests to release branch
+
+After the `release-X.Y` release branch is cut, pull requests (PRs) merged to
+master will be only get released in the next minor release `X.(Y+1).0`.
+
+If you want your PR released earlier in a patch release `X.Y.(Z+1)`:
+* The PR must already get merged to master branch.
+* The PR should be a bug fix.
+* The PR should be cherry picked to corresponding release branch `release-X.Y`.
+
+Choose one of the following options for cherry picking your PR to release branch.
+
+#### Option - (Recommended) cherrypick-approved label
+Contributors should ask OWNERS who approved the PR to add a `cherrypick-approved`
+label if they want the PR cherry picked to release branch.
+
+Release manager will
+* Periodically or before release, search all merged PRs with `cherrypick-approved`
+    label, but no `cherrypicked` label using
+    [this link](https://github.com/kubeflow/pipelines/pulls?q=is%3Apr+label%3Acherrypick-approved+-label%3Acherrypicked+is%3Aclosed+)
+* Use the git cherry-pick option to pick these PR commits into the release branch
+in a batch and add `cherrypicked` label to these PRs.
+
+    NOTE: if there are merge conflicts for a PR, ask the PR author or area OWNER
+    to create a cherry pick PR by themselves following other two options.
+* `git push upstream release-$VERSION` directly to the release branch.
+
+#### Option - git cherry-pick
+* Find the commit you want to cherry pick on master as $COMMIT_SHA.
+* Find the active release branch name $BRANCH, e.g. release-1.0
+*
+    ```bash
+    git co $BRANCH
+    git co -b <cherry-pick-pr-branch-name>
+    git cherry-pick $COMMIT_SHA
+    ```
+* Resolve merge conflicts if any
+* `git push origin HEAD`
+* Create a PR and remember to update PR's destination branch to `release-$VERSION`
+* Ask the same OWNERS that would normally need to approve this PR
+
+#### Option - Kubeflow cherry_pick_pull.sh helper
+Kubeflow has a cherry pick helper script: https://github.com/kubeflow/kubeflow/blob/master/hack/cherry-picks.md
+
+It automates the process using `hub` CLI tool and bash, so it takes some one off efforts to set up for the first time.
+
+After that, this is convenient to do a lot of cherry picks, because for each PR you'd only need to specify
+release branch and PR number.
+
+Known caveats:
+* It may produce PR title that is duplicative, you can edit the title after cherry picking.
+
+## Release Manager Instructions
```

Comment:
@joeliedtke You only need to review content above this line, because the rest are internal release manager facing.